api-tests/src/steps/cri-steps.ts

```diff
@@ -174,3 +178,33 @@ When(
     );
   },
 );
+
+When(
+  /^the subject already has the following (expired )?credentials$/,
+  async function (
+    this: World,
+    expired: "expired " | undefined,
+    table: DataTable,
+  ): Promise<void> {
+    this.userId = this.userId ?? getRandomString(16);
+    const credentials: string[] = [];
+    for (const row of table.hashes()) {
```

Comment:
If you wanted to speed this up, you could do this in parallel with `Promise.all` - but there's no real need unless it's particularly slow (and perhaps harder to read/debug)

```
const credentials = await Promise.all(table.hashes()
  .map(async (row) => criStubClient.generateVc(
    row.CRI,
    await generateVcRequestBody(
      this.userId,
      row.CRI,
      row.scenario,
      expired ? EXPIRED_NBF : undefined,
    ),
  ),
));
```
It'll kick off all the promises in parallel and then wait for them all to complete

Reply:
An easier to follow example without the loop would be:
```
// Await in parallel
const firstPromise = doAsyncThing();
const secondPromise = doAsyncThing();
const results = await Promise.all([firstPromise, secondPromise]);
```
vs.
```
// Await in series
const firstResult = await doAsyncThing();
const secondResult = await doAsyncThing();
const results = [firstResult, secondResult];
```